packages/core/src/modules/oob/OutOfBandModule.ts

```diff
@@ -622,19 +625,34 @@ export class OutOfBandModule {
 
     this.logger.debug(`Message with type ${plaintextMessage['@type']} can be processed.`)
 
+    let serviceEndpoint: string | undefined
+    let recipientKeys: string[] | undefined
+    let routingKeys: string[] = []
+
     // The framework currently supports only older OOB messages with `~service` decorator.
     // TODO: support receiving messages with other services so we don't have to transform the service
     // to ~service decorator
     const [service] = services
 
     if (typeof service === 'string') {
-      throw new AriesFrameworkError('Dids are not currently supported in out-of-band invitation services attribute.')
+      const didDocument = await this.didResolverService.resolveDidDocument(service)
+      const documentService = didDocument?.service?.[0]
+      serviceEndpoint = documentService?.serviceEndpoint
+      recipientKeys = didDocument?.recipientKeys?.map((key) => key.publicKeyBase58)
```

Comment:
This is missing some handling of cases (routingKeys, resolving recipient keys). Maybe we can extract the logic from this method into smaller methods and reuse that in both cases: https://github.com/Iskander508/aries-framework-javascript/blob/oob_public_did/packages/core/src/modules/dids/services/DidResolverService.ts#L79-L134

Reply:
changed to reuse the `resolveServicesFromDid` method